File: content/snippets/go/design_patterns/data_mapper.go
package main

import "fmt"

type User struct {
	ID   int
	Name string
}

func mapRow(row map[string]string) User {
	return User{ID: atoi(row["id"]), Name: row["name"]}
}
func atoi(s string) int { var v int; fmt.Sscan(s, &v); return v }
func main() {
	row := map[string]string{"id": "3", "name": "Carol"}
	u := mapRow(row)
	fmt.Println(u.ID, u.Name)
}

File: content/snippets/python/design_patterns/data_mapper.py
class User:
    def __init__(self, id:int, name:str):
        self.id = id; self.name = name

def map_row(row: dict) -> User:
    return User(int(row['id']), row['name'])

if __name__ == '__main__':
    row = {'id':'2','name':'Bob'}
    u = map_row(row)
    print(u.id, u.name)
